src/Web/Controllers/AccountController.cs

```diff
@@ -206,11 +300,165 @@ public async Task<IActionResult> Register(RegisterViewModel model, string return
 
             if (ModelState.ErrorCount > 0)
             {
-                return View(model);
+                return View("Register", model);
             }
 
             #region Fraud Protection Service
+            var correlationId = _fraudProtectionService.NewCorrelationId;
+
             // Ask Fraud Protection to assess this signup/registration before registering the user in our database, etc.
+            if (useAP)
+            {
+                AccountProtection.SignUp signupEvent = CreateSignupAPEvent(model);
+
+                var signupAssessment = await _fraudProtectionService.PostSignupAP(signupEvent, correlationId);
+
+                //Track Fraud Protection request/response for display only
+                var fraudProtectionIO = new FraudProtectionIOModel(signupEvent, signupAssessment, "Signup");
+                TempData.Put(FraudProtectionIOModel.TempDataKey, fraudProtectionIO);
+
+                //2 out of 3 signups will succeed on average. Adjust if you want more or less signups blocked for tesing purposes.
```

Comment:
AP has sign up rules, so the AP portal of sign up should respect the APIs decision rather than do anything random.  Ditto for sign in.

Reply:
good catch, fixed